pkg/util/hkdf/hkdf.go

```diff
@@ -60,7 +61,16 @@ func expandLabel(secret []byte, label string, context []byte, length int) []byte
 		b.AddBytes(context)
 	})
 	out := make([]byte, length)
-	transcript := crypto.SHA256 // TODO  fixme : use cipher_id argument
+
+	var transcript crypto.Hash
```

Comment:
why do not use `cipher_id` argument?

Reply:
I would say that adding parameters to the method feels a bit redundant, especially doing `cipher_id` bitwise operation twice. The kind of `crypto.Hash` can be easily deduced from the `size` parameter. But I am not a Golang expert, I am not sure the best practice to do this. If you insist, I'll go with your way.

Reply:
I was too busy last week. I will open another PR to fix this later